content/news/2023-10-21-bevy-0.12/index.md

```diff
@@ -15,9 +15,351 @@ Since our last release a few months ago we've added a _ton_ of new features, bug
+* **Transformation**: Some "asset source files" aren't in the right format by default. You can have an asset of type `A` and transform it into type `B`.
+
+If you are building an app that tests the limits of your hardware with optimal formats ... or you just want to cut down on startup / loading times, asset preprocessing is for you.
+
+For an in-depth technical breakdown of the implementation we chose, check out the [Bevy Asset V2 pull request](https://github.com/bevyengine/bevy/pull/8624).
+
+### Enabling Pre-Processing
+
+To enable asset pre-processing, just configure your [`AssetPlugin`] like this:
+
+```rust
+app.add_plugins(DefaultPlugins.set(
+    AssetPlugin {
+        mode: AssetMode::Processed,
+        ..default()
+    }
+))
+```
+
+This will configure the asset system to look for assets in the `imported_assets` folder instead of the `assets` "source folder". During development, enable the `asset_processor` cargo feature flag like this:
+
+```sh
+cargo run --features bevy/asset_processor
+```
+
+This will start the [`AssetProcessor`] in parallel with your app. It will run until all assets are read from their source (defaults to the `assets` folder), processed, and the results have been written to their destination (defaults to the `imported_assets` folder). This pairs with asset hot-reloading. If you make a change, this will be detected by the [`AssetProcessor`], the asset will be reprocessed, and the result will be hot-reloaded in your app.
+
+[`AssetPlugin`]: https://dev-docs.bevyengine.org/bevy/asset/struct.AssetPlugin.html
+[`AssetProcessor`]: https://dev-docs.bevyengine.org/bevy/asset/processor/struct.AssetProcessor.html
+
+### Incremental and Dependency Aware
+
+**Bevy Asset V2** will only process assets that have changed. To accomplish this, it computes and stores hashes of each asset source file:
+
+```rust
+hash: (132, 61, 201, 41, 85, 80, 72, 189, 132, 81, 252, 156, 4, 227, 196, 207),
+```
+
+It also tracks the asset dependencies used when processing an asset. If a dependency has changed, the dependant asset will also be re-processed!
+
+### Transactional and Reliable
+
+**Bevy Asset V2** uses write-ahead logging (a technique commonly used by databases) to recover from crashes / forced exists. Whenever possible it avoids full-reprocessing and only reprocesses incomplete transactions.
+
+The [`AssetProcessor`] can close (either intentionally or unintentionally) at any point in time and it will pick up right where it left off!
+
+If a Bevy App asks to load an asset that is currently being processed (or re-processed), the load will asynchronously wait until both the processed asset and its meta file have been written. This ensures that a loaded asset file and meta file always "match" for a given load.
+
+### Asset Meta Files
+
+Assets now support (optional) `.meta` files. This enables configuring things like:
+
+* **The asset "action"**
```

Comment:
This explanation could be clearer. It's hard to understand what this might be before reading the examples, and even then it requires some logical jumps.

Reply:
Good call!